### ISSUE TITLE
False Positive: NullCoalescingOperatorCanBeUsedInspection

### ISSUE THREAD

**Issue by spacemonkey1337** [kalessil/phpinspectionsea#1513]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended)  |
| Language level | PHP 7.0                         |

Current behaviour
--
This sample code:
```php
$item = ucfirst($item);

if (isset($map[$item])) {
    $item = $map[$item];
}
```
Being suggested to replace with:
```php
$item = $map[$item] ?? ucfirst($item);
```
This is not accurate since `$map[$item]` need to be accessed after `$item` transformation. So actually it's a `$map[ucfirst($item)]`.

Expected behaviour
--
1. To not suggest null coalescing pattern for such cases.
2. To suggest correct (but not pretty) replacement:
```php
$item = $map[ucfirst($item)] ?? ucfirst($item);
```

Environment details
---
PhpStorm 2019.3.2

**Comment by kalessil** [comment]
Thank you for reporting @spacemonkey1337, I'll take care of fixing this.

**Comment by kalessil** [comment]
Fixed!
